tutorial/pytorch_agent/train_cpu_model-pytorch.py

```diff
@@ -110,96 +137,218 @@ def main():
+
+    for epoch in range(EPOCH_SIZE):
+        loss, accuracy = train(model, optimizer, train_loader, val_loader)
+
+        with tune.checkpoint_dir(epoch) as checkpoint_dir:
+            path = os.path.join(checkpoint_dir, "checkpoint")
+            torch.save((model.state_dict(), optimizer.state_dict()), path)
+
+        # Send the current training result back to Tune
+        tune.report(loss=loss, accuracy=accuracy)
+
+def train(model, optimizer, train_loader, val_loader):
+    loss_fn = nn.MSELoss()
+
+    for idx, (inputs, target_control) in enumerate(train_loader):
+        model.train()
+        # Clear gradient
+        optimizer.zero_grad()
+        # Get model output
+        predicted_control = model(inputs)
+        # loss calculation vs target
+        loss = loss_fn(predicted_control, target_control)
+        loss.backward()
+        # update model weights
+        optimizer.step()
+
+    model.eval()
+    prediction_total = 0
+    prediction_correct = 0
+    val_loss = 0
+    val_steps = 0
+    with torch.no_grad():
+        for idx, (inputs, target_control) in enumerate(val_loader):
+            prediction_total += inputs.size(0)
+
+            # Run inputs through model, save prediction
+            predicted_control = model(inputs)
+            # Round to nearest 100 MHz increment
+            predicted_control = np.round(predicted_control,1)
+            prediction_correct += (target_control == predicted_control).sum().item()
+
+            loss = loss_fn(predicted_control, target_control)
+            val_loss += loss.numpy()
+            val_steps += 1;
+
+
+    return val_loss/val_steps, prediction_correct/prediction_total
+
+def test(model, test_loader):
+    prediction_total = 0
+    prediction_correct = 0
+    prediction_min = 0
+    prediction_max = 0
+    with torch.no_grad():
+        for idx, (inputs, target_control) in enumerate(test_loader):
+            prediction_total += inputs.size(0)
+            # Run inputs through model, save prediction
+            predicted_control = model(inputs)
+            # Round to nearest 100 MHz increment
+            predicted_control = np.round(predicted_control,1)
+
+            # Generate stats
+            prediction_correct += (target_control == predicted_control).sum().item()
+
+            if (idx == 0):
+                prediction_max = predicted_control.max()
+                prediction_min = predicted_control.min()
+            else:
+                if (prediction_max < predicted_control.max()):
+                    prediction_max = predicted_control.max()
+                if (prediction_min > predicted_control.min()):
+                    prediction_min = predicted_control.min()
+
+        print('\ttest output min: {:.2f}, max: {:.2f}.'.format(prediction_min, prediction_max))
+    return prediction_correct/prediction_total
+
+class P3Net(nn.Module):
+    def __init__(self, width_input, width_fc, depth_fc):
+        super(P3Net, self).__init__()
```

Comment:
Maybe related to the error message you're seeing: Why not `super().__init__()`?

Reply:
super().__init__() may work here as well.  In testing it would occasionally hit the same error however.